src/ResourceManager/Profile/Commands.Profile/Account/AddAzureRmAccount.cs

```diff
@@ -50,56 +52,60 @@ public class AddAzureRMAccountCommand : AzureRMCmdlet, IModuleAssemblyInitialize
 
         public string EnvironmentName { get; set; }
 
-
-        [Parameter(ParameterSetName = UserParameterSet, Mandatory = false, HelpMessage = "Optional credential")]
-        [Parameter(ParameterSetName = ServicePrincipalParameterSet, Mandatory = true, HelpMessage = "Credential")]
-        [Parameter(ParameterSetName = SubscriptionIdParameterSet, Mandatory = false, HelpMessage = "Optional credential")]
-        [Parameter(ParameterSetName = SubscriptionNameParameterSet, Mandatory = false, HelpMessage = "Optional credential")]
+        
+        [Parameter(ParameterSetName = UserParameterSetWithSubscriptionId, Mandatory = false, HelpMessage = "Optional credential", Position = 0)]
```

Comment:
Note: if there wasn't another change to make I would not make this comment.  Please break the Parameter attributes over multiple lines so they are not so difficult to read in the default code review pane.